_component_design/bulk-selection.md

```diff
@@ -0,0 +1,208 @@
+preview-image: preview-images/tables-controls.svg
+status: stable
+resource: true
+last-modified: 2018-12-11
+helix-ui-css: false
+helix-ui-javascript: false
+---
+
+{% include toc.html %}
+
+<section class="static-section" markdown="1">
+
+<div class="hxRow"  markdown="1">
+
+{% column left:"hxCol hxSpan-12-xs hxSpan-12-sm hxSpan-8-md hxSpan-8-lg" %}
+
+## When to use
+
+Add bulk selection options to a table when you want users to be able to select (or deselect) all visible rows, and then perform an action on the group of selected rows. If the information you are presenting in a table is read-only, do not provide bulk selection.
+
+When compared to selecting checkboxes individually, bulk selection is a more convenient approach to selecting or deselecting all visible rows.
+
+{% endcolumn %}
+
+</div>
+
+</section>
+
+<section class="static-section" markdown="1">
+
+<div class="hxRow"  markdown="1">
+
+{% column left:"hxCol hxSpan-12-xs hxSpan-12-sm hxSpan-8-md hxSpan-8-lg" %}
+
+## Best practices
+
+Consider the following best practices when you add bulk selection to a table:
+
+- The table should support the keyboard shortcuts to selecting multiple objects. For example, after selecting a row, the user can hold down the **Shift** key and select another row. This action selects all rows between the first and second selections.
+- Construct the selection bucket so that the **Deselect All** option launches a confirmation modal that asks the user to confirm their choice.
+- xxxxxx.
+- xxxxxx.
+- xxcxxx.
+
+{% endcolumn %}
+
+</div>
+
+</section>
+
+<section class="static-section" markdown="1">
+
+<div class="hxRow"  markdown="1">
+
+{% column left:"hxCol hxSpan-12-xs hxSpan-12-sm hxSpan-8-md hxSpan-8-lg" %}
+
+## Specifications
+
+Use the following specifications when you construct bulk selection.
+
+{% endcolumn %}
+
+</div>
+
+</section>
+
+<section class="static-section" markdown="1">
+
+<div class="hxRow"  markdown="1">
+
+{% column left:"hxCol hxSpan-12-xs hxSpan-12-sm hxSpan-4-md hxSpan-4-lg" %}
+
+### Composition
+
+Bulk selection contains the following elements:
+
+- **Parent checkbox (required)**: When the user selects the parent checkbox, all children checkboxes are selected. When the user deselects the parent checkbox, all children checkboxes are deselected.
+- **Select All (optional)**: When the user clicks **Select All**, all records on the current page are selected.
+- **Deselect All (optional)**: When the user click **Deselect All**, all records on the current page are deselected.
+- **Number selected (optional)**: The number selected shows the number of objects selected in the table. For example, **3 Selected** means that out of 10 records displayed in the table, three of those records are currently selected by the user. If you include number selected, then also include the **Select All** option.
```

Comment:
include the select all or deselect all depending on your workflow.

Then remove select all and deselect all from composition.

Reply:
Done.